sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -175,13 +176,49 @@ case class DescribeDatabase(
   }
 }
 
+/**
+ * The DDL command that creates a function.
+ * alias: the class name that implements the created function.
+ * resources: Jars, files, or archives which need to be added to the environment when the function
+ *            is referenced for the first time by a session.
+ * isTemp: indicates if it is a temporary function.
+ */
 case class CreateFunction(
     databaseName: Option[String],
     functionName: String,
     alias: String,
     resources: Seq[(String, String)],
     isTemp: Boolean)(sql: String)
-  extends NativeDDLCommand(sql) with Logging
+  extends NativeDDLCommand(sql) with Logging {
+
+  override def run(sqlContext: SQLContext): Seq[Row] = {
+    val func = FunctionIdentifier(functionName, databaseName)
+    val catalogFunc = CatalogFunction(func, alias, resources)
+    if (isTemp) {
+      // Set `ignoreIfExists` to false, so if the temporary function already exists,
+      // an exception will be thrown.
+      val (info, builder) =
+        sqlContext.sessionState.functionRegistry.makeFunctionBuilderAndInfo(functionName, alias)
+      sqlContext.sessionState.catalog.createTempFunction(functionName, info, builder, false)
+    } else {
+      // Check if the function to create is already existing. If so, throw exception.
+      var funcExisting: Boolean =
```

Comment:
actually I would push this into the session catalog itself and implement a `functionExists` method there